gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java

```diff
@@ -28,46 +28,51 @@
 class QueryResultsImpl<T> extends AbstractIterator<T> implements QueryResults<T> {
 
   private final DatastoreImpl datastore;
-  private final DatastoreV1.ReadOptions readOptionsPb;
-  private final DatastoreV1.PartitionId partitionIdPb;
+  private final com.google.datastore.v1beta3.ReadOptions readOptionsPb;
+  private final com.google.datastore.v1beta3.PartitionId partitionIdPb;
   private final ResultType<T> queryResultType;
   private Query<T> query;
   private ResultType<?> actualResultType;
-  private DatastoreV1.QueryResultBatch queryResultBatchPb;
+  private com.google.datastore.v1beta3.QueryResultBatch queryResultBatchPb;
   private boolean lastBatch;
-  private Iterator<DatastoreV1.EntityResult> entityResultPbIter;
-  //private ByteString cursor; // only available in v1beta3
+  private Iterator<com.google.datastore.v1beta3.EntityResult> entityResultPbIter;
+  private ByteString cursor;
 
 
-  QueryResultsImpl(DatastoreImpl datastore, DatastoreV1.ReadOptions readOptionsPb,
+  QueryResultsImpl(DatastoreImpl datastore, com.google.datastore.v1beta3.ReadOptions readOptionsPb,
                    Query<T> query) {
     this.datastore = datastore;
     this.readOptionsPb = readOptionsPb;
     this.query = query;
     queryResultType = query.type();
-    DatastoreV1.PartitionId.Builder pbBuilder = DatastoreV1.PartitionId.newBuilder();
-    pbBuilder.setDatasetId(datastore.options().projectId());
+    com.google.datastore.v1beta3.PartitionId.Builder pbBuilder =
+        com.google.datastore.v1beta3.PartitionId.newBuilder();
+    pbBuilder.setProjectId(datastore.options().projectId());
     if (query.namespace() != null) {
-      pbBuilder.setNamespace(query.namespace());
+      pbBuilder.setNamespaceId(query.namespace());
     } else if (datastore.options().namespace() != null) {
-      pbBuilder.setNamespace(datastore.options().namespace());
+      pbBuilder.setNamespaceId(datastore.options().namespace());
     }
     partitionIdPb = pbBuilder.build();
     sendRequest();
+    if (queryResultBatchPb.getSkippedResults() > 0) {
+      cursor = queryResultBatchPb.getSkippedCursor();
+    } else {
+      cursor = ByteString.EMPTY;
```

Comment:
@aozarov this is fixed now.  The default value for a `ByteString` is an empty `ByteString` in proto3, so cases (2) and (3) are now handled in the `else`